### ISSUE TITLE
Implement Retention Policy in External Event

### ISSUE THREAD

**Issue by sauljabin** [littlehorse-enterprises/littlehorse#562]
### Context

PENDING

### Acceptance Criteria

PENDING

**Scenario** [the name for the behavior that will be described]
**Given** [the beginning state of the scenario]
**When** [specific action that the user makes]
**Then** [the outcome of the action in **When**]
**And** [used to continue any of three previous statements]

### Components

- [ ] Dashboard
- [X] Server
- [X] Python SDK
- [X] Go SDK
- [X] Java SDK
- [X] C# SDK
- [ ] LH Control
- [ ] LH Tests Utils

**Comment by coltmcnealy-lh** [comment]
I think I had actually already implemented this ticket. However, the `lhctl get externalEvent` is not working. I willl need to debug that

**Comment by coltmcnealy-lh** [comment]
Resolution:
* The `PutExternalEventDefModel#process()` was **not** saving the `retentionPolicy` properly on the new `ExternalEventDef`. That is fixed.
* The recent changes to `lhctl` caused segmentation faults when the tenant id was not set. That has been fixed.
* A pull request from earlier made a breaking change to the `ExternalEvent` proto which is why the `lhctl get externalEvent` didn't work. Once I updated `lhctl` to the new version via `go install ./lhctl`, then it has worked.
